tests/i18n/override.py

```diff
@@ -0,0 +1,168 @@
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) == 'gl'
+
+    # Test cookie with longer supported language.
+    request.COOKIES[settings.LANGUAGE_COOKIE_NAME] = 'en-US'
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) == 'en-US'
+
+    # Test cookie with longer underscore language code for a supported
+    # language.
+    request.COOKIES[settings.LANGUAGE_COOKIE_NAME] = 'gl_ES'
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) is None
+
+    # Test cookie with longer hyphen language code for a supported language.
+    request.COOKIES[settings.LANGUAGE_COOKIE_NAME] = 'fr-FR'
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) is None
+
+    # Test cookie with shorter language code for a supported language.
+    request.COOKIES[settings.LANGUAGE_COOKIE_NAME] = 'en'
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) is None
+
+    # Test cookie with unsupported language.
+    request.COOKIES[settings.LANGUAGE_COOKIE_NAME] = 'FAIL'
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) is None
+
+
+def test_get_lang_from_http_header(rf):
+    # Test no header.
+    request = rf.get("")
+    assert 'HTTP_ACCEPT_LANGUAGE' not in request.META
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test empty header.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with wildcard.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='*')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='gl')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) == 'gl'
+
+    # Test cookie with longer supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='en-US')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with longer underscore language code for a supported
+    # language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='gl_ES')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with longer hyphen language code for a supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='fr-FR')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) == 'fr'
+
+    # Test header with shorter language code for a supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='en')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with unsupported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='FAIL')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+
+@pytest.mark.django_db
+def test_get_lang_from_session(rf, client):
+    # Test no session.
+    request = rf.get("")
+    assert not hasattr(request, 'session')
```

Comment:
i think this assert is not what is being tested - better add a comment to say that there is no session

Reply:
Will do